onnxruntime/core/providers/qnn/qnn_execution_provider.cc

```diff
@@ -325,16 +325,6 @@ QNNExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_viewer
   if (supported_nodes.empty()) {
     LOGS(logger, INFO) << "Number of partitions supported by QNN EP: 0";
     return result;
-  } else if (supported_nodes.size() == 1) {
-    const auto* node = *supported_nodes.begin();
-    if (node->OpType() == "QuantizeLinear" || node->OpType() == "DequantizeLinear") {
```

Comment:
This tried to filter out cases where QNN EP could only handle a single Q/DQ node in the graph. However, it doesn't account for the possibility that multiple partitions may have a single Q/DQ node.

Since we're already looping through the partitions to count the number of supported nodes, I added a filtering step (see below) that does not keep partitions with a single Q/DQ node.